mssqlcli/mssqlcliclient.py

```diff
@@ -23,83 +23,143 @@ def generate_owner_uri():
-            return self.owner_uri, []
+        logger.info(u'Initialized MssqlCliClient with owner Uri {}'.format(self.owner_uri))
 
-        # Required params
-        connection_params = {u'ServerName': self.server_name,
-                             u'DatabaseName': self.database,
-                             u'UserName': self.user_name,
-                             u'Password': self.password,
-                             u'AuthenticationType': self.authentication_type,
-                             u'OwnerUri': self.owner_uri,
-                             u'MultipleActiveResultSets': self.multiple_active_result_sets}
+    def get_base_connection_params(self):
+        return {u'ServerName': self.server_name,
+                u'DatabaseName': self.database,
+                u'UserName': self.user_name,
+                u'Password': self.password,
+                u'AuthenticationType': self.authentication_type,
+                u'OwnerUri': self.owner_uri
+                }
 
-        # Optional params
+    def add_optional_connection_params(self, base_connection_params):
         if self.encrypt:
-            connection_params[u'Encrypt'] = self.encrypt
+            base_connection_params[u'Encrypt'] = self.encrypt
         if self.trust_server_certificate:
-            connection_params[u'TrustServerCertificate'] = self.trust_server_certificate
+            base_connection_params[u'TrustServerCertificate'] = self.trust_server_certificate
         if self.connection_timeout:
-            connection_params[u'ConnectTimeout'] = self.connection_timeout
+            base_connection_params[u'ConnectTimeout'] = self.connection_timeout
         if self.application_intent:
-            connection_params[u'ApplicationIntent'] = self.application_intent
+            base_connection_params[u'ApplicationIntent'] = self.application_intent
         if self.multi_subnet_failover:
-            connection_params[u'MultiSubnetFailover'] = self.multi_subnet_failover
+            base_connection_params[u'MultiSubnetFailover'] = self.multi_subnet_failover
         if self.packet_size:
-            connection_params[u'PacketSize'] = self.packet_size
+            base_connection_params[u'PacketSize'] = self.packet_size
 
-        connection_params.update(self.extra_params)
+        base_connection_params.update(self.extra_params)
 
-        connection_request = self.sql_tools_client.create_request(
-            u'connection_request', connection_params, self.owner_uri)
-        connection_request.execute()
+        return base_connection_params
+
+    def connect_to_database(self):
+        connection_params = self.get_base_connection_params()
+        connection_params = self.add_optional_connection_params(connection_params)
+
+        owner_uri, error_messages = self._execute_connection_request_with(connection_params)
+
+        return owner_uri, error_messages
+
+    def execute_multi_statement(self, query):
+        # Try to run first as special command
```

Comment:
From and API standpoint, it would be nice if we had a single public method to execute on or more statements: execute_query. So execute_single_statement would become private. Any downside to this?

Reply:
Agreed, no downsides other then background queries will run through additional code path but it wouldn't be harmful.

---
In reply to: [170340117](https://github.com/dbcli/mssql-cli/pull/170#discussion_r170340117) [](ancestors = 170340117)